### PR TITLE
add CI build and test for context-py

### PR DESCRIPTION
**Pull request by dfrankland**
Adds a build and test process for `context-py` Python bindings

### CODE REVIEW

File: .github/workflows/pyo3.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          container: "2_28"
+          before-script-linux: |
+            # Needed for `openssl-sys`:
+            # https://docs.rs/openssl/latest/openssl/
+            if [ -f /usr/bin/dnf ]; then
+              # quay.io/pypa/manylinux_2_28_x86_64:latest
+              /usr/bin/dnf upgrade
+              /usr/bin/dnf install -y pkgconf perl-IPC-Cmd openssl-devel python3-pip
+              /usr/bin/ln -s /usr/bin/pip3 /usr/bin/pip
+              /usr/bin/ln -sf /usr/bin/python3.11 /usr/bin/python3
```
Comment (gnalh) [inline .github/workflows/pyo3.yml:51]: What is this?

**Review by gnalh — APPROVED** [review]
Overall looks good. Now that we have Python code, can we also add python linters into this repo?

File: .github/workflows/pyo3.yml (hunk 1 above)
Comment (dfrankland) [reply .github/workflows/pyo3.yml:51]:
```
            # Needed for `openssl-sys`:
            # https://docs.rs/openssl/latest/openssl/
```

File: rustfmt.toml
Comment (dfrankland) [inline rustfmt.toml:1]: This file is not used by `trunk` CLI (and IMO is _very_ divergent from normal Rust code styling)

File: .trunk/configs/.bandit [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[bandit]
+exclude = tests
```
Comment (dfrankland) [inline .trunk/configs/.bandit:2]: `assert`s are used in tests and bandit says to exclude `tests` 🤷 
https://bandit.readthedocs.io/en/latest/config.html#exclusions

File: .trunk/trunk.yaml
Comment (dfrankland) [inline .trunk/trunk.yaml:1]: upgraded to support python linters and formatters

File: .trunk/trunk.yaml [hunk 3]
```diff
@@ -7,34 +7,38 @@ cli:
...
+    - trufflehog@3.82.6
     - yamllint@1.35.1
 actions:
-  disabled:
+  enabled:
     - trunk-announce
     - trunk-check-pre-push
     - trunk-fmt-pre-commit
-  enabled:
     - trunk-upgrade-available
```
Comment (dfrankland) [inline .trunk/trunk.yaml:42]: I think we want to use these

**Pull request merged**